[2025-01-28T15:15:43.164+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_price_pipeline_daily.start_consumer_daily manual__2025-01-28T15:15:21.051582+00:00 [queued]>
[2025-01-28T15:15:43.171+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_price_pipeline_daily.start_consumer_daily manual__2025-01-28T15:15:21.051582+00:00 [queued]>
[2025-01-28T15:15:43.172+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2025-01-28T15:15:43.191+0000] {taskinstance.py:1382} INFO - Executing <Task(BashOperator): start_consumer_daily> on 2025-01-28 15:15:21.051582+00:00
[2025-01-28T15:15:43.195+0000] {standard_task_runner.py:57} INFO - Started process 122 to run task
[2025-01-28T15:15:43.199+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'stock_price_pipeline_daily', 'start_consumer_daily', 'manual__2025-01-28T15:15:21.051582+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/stock_pipeline_dag.py', '--cfg-path', '/tmp/tmp24bwl12q']
[2025-01-28T15:15:43.200+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask start_consumer_daily
[2025-01-28T15:15:43.252+0000] {task_command.py:415} INFO - Running <TaskInstance: stock_price_pipeline_daily.start_consumer_daily manual__2025-01-28T15:15:21.051582+00:00 [running]> on host ffc5ba868373
[2025-01-28T15:15:43.836+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_price_pipeline_daily' AIRFLOW_CTX_TASK_ID='start_consumer_daily' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T15:15:21.051582+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-28T15:15:21.051582+00:00'
[2025-01-28T15:15:43.843+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-01-28T15:15:43.844+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'docker exec consumer-daily python kafka_consumer_daily.py']
[2025-01-28T15:15:43.856+0000] {subprocess.py:86} INFO - Output:
[2025-01-28T15:15:45.448+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-01-28T15:15:45.449+0000] {subprocess.py:93} INFO - INFO:kafka.conn:Probing node bootstrap-0 broker version
[2025-01-28T15:15:45.449+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-01-28T15:15:45.557+0000] {subprocess.py:93} INFO - INFO:kafka.conn:Broker version identified as 2.5.0
[2025-01-28T15:15:45.558+0000] {subprocess.py:93} INFO - INFO:kafka.conn:Set configuration api_version=(2, 5, 0) to skip auto check_version requests on startup
[2025-01-28T15:15:45.559+0000] {subprocess.py:93} INFO - INFO:kafka.consumer.subscription_state:Updating subscribed topics to: ('stock_prices_daily',)
[2025-01-28T15:15:45.562+0000] {subprocess.py:93} INFO - INFO:kafka.cluster:Group coordinator for stock_pipeline_daily is BrokerMetadata(nodeId='coordinator-1001', host='kafka', port=9092, rack=None)
[2025-01-28T15:15:45.564+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Discovered coordinator coordinator-1001 for group stock_pipeline_daily
[2025-01-28T15:15:45.564+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Starting new heartbeat thread
[2025-01-28T15:15:45.565+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator.consumer:Revoking previously assigned partitions set() for group stock_pipeline_daily
[2025-01-28T15:15:45.565+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=coordinator-1001 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-01-28T15:15:45.565+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=coordinator-1001 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-01-28T15:15:45.565+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Closing connection.
[2025-01-28T15:15:45.667+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:(Re-)joining group stock_pipeline_daily
[2025-01-28T15:15:48.674+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Elected group leader -- performing partition assignments using range
[2025-01-28T15:15:48.679+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=1001 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-01-28T15:15:48.680+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=1001 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-01-28T15:15:48.685+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Successfully joined group stock_pipeline_daily with generation 1
[2025-01-28T15:15:48.686+0000] {subprocess.py:93} INFO - INFO:kafka.consumer.subscription_state:Updated partition assignment: [TopicPartition(topic='stock_prices_daily', partition=0)]
[2025-01-28T15:15:48.687+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator.consumer:Setting newly assigned partitions {TopicPartition(topic='stock_prices_daily', partition=0)} for group stock_pipeline_daily
[2025-01-28T15:15:48.815+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Stopping heartbeat thread
[2025-01-28T15:15:48.816+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Leaving consumer group (stock_pipeline_daily).
[2025-01-28T15:15:48.827+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=coordinator-1001 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Closing connection.
[2025-01-28T15:15:48.827+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection node_id=1001 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Closing connection.
[2025-01-28T15:15:48.827+0000] {subprocess.py:93} INFO - INFO:__main__:Consumer closed.
[2025-01-28T15:15:48.828+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-01-28T15:15:48.830+0000] {subprocess.py:93} INFO -   File "/app/kafka_consumer_daily.py", line 35, in process_and_store
[2025-01-28T15:15:48.831+0000] {subprocess.py:93} INFO -     symbol = data.get("Meta Data").get("2. Symbol")
[2025-01-28T15:15:48.831+0000] {subprocess.py:93} INFO - AttributeError: 'NoneType' object has no attribute 'get'
[2025-01-28T15:15:48.832+0000] {subprocess.py:93} INFO - 
[2025-01-28T15:15:48.832+0000] {subprocess.py:93} INFO - During handling of the above exception, another exception occurred:
[2025-01-28T15:15:48.832+0000] {subprocess.py:93} INFO - 
[2025-01-28T15:15:48.833+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-01-28T15:15:48.834+0000] {subprocess.py:93} INFO -   File "/app/kafka_consumer_daily.py", line 99, in <module>
[2025-01-28T15:15:48.834+0000] {subprocess.py:93} INFO -     main()
[2025-01-28T15:15:48.835+0000] {subprocess.py:93} INFO -   File "/app/kafka_consumer_daily.py", line 88, in main
[2025-01-28T15:15:48.836+0000] {subprocess.py:93} INFO -     process_and_store(message.value)
[2025-01-28T15:15:48.836+0000] {subprocess.py:93} INFO -   File "/app/kafka_consumer_daily.py", line 73, in process_and_store
[2025-01-28T15:15:48.836+0000] {subprocess.py:93} INFO -     logger.error(f"Error processing and storing data for symbol {symbol}: {e}")
[2025-01-28T15:15:48.837+0000] {subprocess.py:93} INFO - UnboundLocalError: local variable 'symbol' referenced before assignment
[2025-01-28T15:15:49.066+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-01-28T15:15:49.078+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 210, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-01-28T15:15:49.084+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_price_pipeline_daily, task_id=start_consumer_daily, execution_date=20250128T151521, start_date=20250128T151543, end_date=20250128T151549
[2025-01-28T15:15:49.110+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task start_consumer_daily (Bash command failed. The command returned a non-zero exit code 1.; 122)
[2025-01-28T15:15:49.151+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-01-28T15:15:49.171+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
